pkg/azure/api/providerspec.go

```diff
@@ -0,0 +1,265 @@
+/*
+SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package api defined the schema of the Azure Provider Spec
+package api
+
+const (
+	// AzureClientID is a constant for a key name that is part of the Azure cloud credentials.
+	// Deprecated: Use ClientID instead.
+	AzureClientID string = "azureClientId"
+	// AzureClientSecret is a constant for a key name that is part of the Azure cloud credentials.
+	// Deprecated: Use ClientSecret instead
+	AzureClientSecret string = "azureClientSecret"
+	// AzureSubscriptionID is a constant for a key name that is part of the Azure cloud credentials.
+	// Deprecated: Use SubscriptionID instead
+	AzureSubscriptionID string = "azureSubscriptionId"
+	// AzureTenantID is a constant for a key name that is part of the Azure cloud credentials.
+	// Deprecated: Use TenantID instead
+	AzureTenantID string = "azureTenantId"
+
+	// AzureAlternativeClientID is a constant for a key name of a secret containing the Azure credentials (access id).
+	// Deprecated: Use ClientID instead.
+	AzureAlternativeClientID = "clientID"
+	// AzureAlternativeClientSecret is a constant for a key name of a secret containing the Azure credentials (access
+	// secret).
+	// Deprecated: Use ClientSecret instead
+	AzureAlternativeClientSecret = "clientSecret"
+	// AzureAlternativeSubscriptionID is a constant for a key name of a secret containing the Azure credentials
+	// (subscription id).
+	// Deprecated: Use ClientID instead.
+	AzureAlternativeSubscriptionID = "subscriptionID"
+	// AzureAlternativeTenantID is a constant for a key name of a secret containing the Azure credentials (tenant id).
+	// Deprecated: Use TenantID instead
+	AzureAlternativeTenantID = "tenantID"
+
+	// ClientID is a constant for a key name that is part of the Azure cloud credentials.
+	ClientID string = "clientID"
+	// ClientSecret is a constant for a key name that is part of the Azure cloud credentials.
+	ClientSecret string = "clientSecret"
+	// SubscriptionID is a constant for a key name that is part of the Azure cloud credentials.
+	SubscriptionID string = "subscriptionID"
+	// TenantID is a constant for a key name that is part of the Azure cloud credentials.
+	TenantID string = "tenantID"
+
+	// MachineSetKindAvailabilitySet is the machine set kind for AvailabilitySet.
+	// Deprecated. Use AzureVirtualMachineProperties.AvailabilitySet instead.
```

Comment:
pls do similar for other occurrences 
```suggestion
	// Deprecated: Use AzureVirtualMachineProperties.AvailabilitySet instead.
```

Reply:
done